sound/soc/sof/intel/telemetry.h

```diff
@@ -0,0 +1,210 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * telemetry data in debug windows
+ */
+
+#ifndef _SOF_INTEL_TELEMETRY_H
+#define _SOF_INTEL_TELEMETRY_H
+
+#define SOF_IPC4_FW_REPORTED_MAX_CORES_COUNT 5
+#define SOF_IPC4_FW_MAX_REPORTED_TASKS 40
+#define SOF_IPC4_FW_AREGS_COUNT 64
+#define SOF_IPC4_FW_MAX_NUMBER_MEM_BINS 1
+#define SOF_IPC4_FW_LP_TIMER_DELAY_CAUSE_COUNT 2
+#define SOF_IPC4_FW_MAX_CONDITION_DATA_COUNT 9
+
+struct system_tick_info_s {
+	u32 count;
+	u32 last_sys_tick_count;
+	u32 max_sys_tick_count;
+	u32 last_ccount;
+	u32 avg_utilization;
+	u32 peak_utilization;
+	u32 peak_utilization_4k;
+	u32 peak_utilization_8k;
+	u32 rsvd[2];
+};
+
+struct deadlock_info_s {
+	u32 register_a0;
+	u32 register_a1;
+	u32 cached_stack_ptr;
+	/* caller, params, condition and return value  for deadlock */
+	u32 run_condition_data[SOF_IPC4_FW_MAX_CONDITION_DATA_COUNT];
+};
+
+struct assert_statistics_s {
+	u32 asserts_triggered_count;
+	u32 last_assert_log_id;
+};
+
+struct assert_info_s {
+	u32 mode;
+	struct assert_statistics_s stats[SOF_IPC4_FW_REPORTED_MAX_CORES_COUNT];
+};
+
+struct xxxruns_info_s {
+	u32 gateway_underruns_count;
+	u32 last_underrun_gateway_node_id;
+	u32 gateway_overruns_count;
+	u32 last_overrun_gateway_node_id;
+	u32 mixer_underruns_count;
+	u32 last_mixer_underrun_resource_id;
+};
+
+struct performance_info_s {
+	u32 cpu_frequency;
+	u32 dropped_lp_timer_ticks_cause[SOF_IPC4_FW_LP_TIMER_DELAY_CAUSE_COUNT];
+};
+
+struct sram_mem_info_s {
+	u32 total_pages_cnt;
+	u32 pages_in_use_cnt;
+	u32 phys_pages_in_use_cnt;
+};
+
+struct sram_mem_pool_info_s {
+	u32 registered_pools_cnt;
+	u32 pools_in_use_cnt;
+	u32 registered_pages_cnt;
+	u32 pages_in_use_cnt;
+};
+
+struct sram_oma_mem_pool_info_s {
```

Comment:
what does 'oma' stand for? it's 'grandmother' in German, I don't know what it could possibly relate to. a cpmment wouldn't hurt.

Reply:
opportunistic memory.  I can't link oma to this words